src/utils/errors.ts

```diff
@@ -24,11 +24,42 @@ export const logError = (error: unknown, context?: Record<string, unknown>) => {
     message:
       error instanceof Error ? error.message : "An unknown error occurred",
     stack: error instanceof Error ? error.stack : undefined,
-    details: error instanceof AppError ? error.details : undefined,
+    details:
+      error instanceof AppError
+        ? error.details
+        : error instanceof Error
+          ? Object.getOwnPropertyNames(error).reduce<Record<string, unknown>>(
```

Comment:
We can now configure this through pino, and automatically add/redact fields more generally. Not just in the middleware, but at the level of the logger.